internal/cluster/common.go

```diff
@@ -272,7 +272,7 @@ func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr []strin
 		previousSecondaryMachineCidr = network.GetMachineCidrById(cluster, 1)
 	}
 
-	if machineCidr[0] != previousPrimaryMachineCidr || (len(machineCidr) > 1 && machineCidr[1] != previousSecondaryMachineCidr) {
+	if (len(machineCidr) > 0 && machineCidr[0] != previousPrimaryMachineCidr) || (len(machineCidr) > 1 && machineCidr[1] != previousSecondaryMachineCidr) {
```

Comment:
Maybe it is a good opportunity to call it machineNetworks and nor machineCidr ?

Reply:
It is an existing code. I'd like not to change it in this PR. We can open a different one for fixing this function including the transaction issue